### PR TITLE
Fix importing and reselecting the same account and cleanup

### PR DESCRIPTION
**Pull request by lukaw3d**
Fixes https://github.com/oasisprotocol/oasis-wallet-web/pull/980#issuecomment-1238192565

### CODE REVIEW

**Comment by buberdds** [comment]
re-importing of a wallet does not work without `walletSelected`. if we want to get rid of it we should handle that in `selectWallet`

**Comment by lukaw3d** [comment]
I'm not sure what you mean. Yes, re-importing of a wallet does not work, either way. This is not an attempt to fix it tho.

And an ugly way to fix it remains possible:
```ts
  if (selectImmediately) {
    yield* put(walletActions.selectWallet(undefined))
    yield* put(walletActions.selectWallet(walletId))
  }
```

**Comment by buberdds** [comment]
Yes, please include this action if you want to remove `walletSelected`.

**Comment by lukaw3d** [comment]
Added back selecting `undefined` as a quickfix.

Larger solutions:
- when importing: disable checkboxes for existing accounts
- remove useRouteRedirects and redirect in import components

File: src/app/state/wallet/saga.ts [hunk 1]
```diff
@@ -129,15 +127,17 @@ export function* openWalletFromMnemonic({ payload: mnemonic }: PayloadAction<str
...
   const existingWallet = yield* call(getWalletByAddress, newWallet.address)
   if (!existingWallet) {
     yield* put(walletActions.walletOpened(newWallet))
   }
 
   const walletId = existingWallet ? existingWallet.id : newWallet.id
 
-  if (newWallet.selectImmediately) {
+  if (selectImmediately) {
+    yield* put(walletActions.selectWallet(undefined)) // Workaround so useRouteRedirects detects selecting the same account
```
Comment (buberdds) [inline src/app/state/wallet/saga.ts:140]: Why you have restored this line in a different place?

Comment (lukaw3d) [reply src/app/state/wallet/saga.ts:140]: To avoid bugs with importing multiple accounts

**Comment by buberdds** [comment]
not sure it's wip or ready for review, but reselecting is still failing for me for a mnemonic and ledger, but works for a priv key.

**Comment by lukaw3d** [comment]
oh, it's supposed to work :/

**Comment by lukaw3d** [comment]
Alright, it's ready now.
I'll simplify into these commits after review:
```
Remove unused walletLoaded action
Deduplicate walletSelected and selectWallet
Don't keep selectImmediately in wallet state
Deduplicate selecting wallet by using selectImmediately
Test importing multiple accounts
Test reimporting and reselecting the same accounts
Fix importing and reselecting the same account
```

File: src/app/state/wallet/saga.ts [hunk 2]
```diff
@@ -114,29 +107,30 @@ export function* openWalletFromMnemonic() {
...
+    yield* delay(1)
```
Comment (buberdds) [inline src/app/state/wallet/saga.ts:133]: ouch, can you add a comment that this is used to avoid React batch updates?

**Review by buberdds — APPROVED** [review]

File: src/app/state/wallet/saga.ts (hunk 2 above)
Comment (lukaw3d) [reply src/app/state/wallet/saga.ts:133]: yep yep

Tho I'm not sure why this would be batched in react 17 :shrug:

**Pull request merged**